### DISCUSSION TITLE
is it possible to rename a field in a ParamObject?

### DISCUSSION THREAD

**Discussion by joprice** [fable-compiler/Fable#3859] [Q&A]
With a param object like https://fable.io/docs/javascript/features.html#paramobject, is it possible to bind a field to a different name in the js? I tried using `CompiledName` and `Emit`, but I don't think these attributes are being used for determining the object's fields.

**Comment by MangelMaxime** [comment]
No the name of the field is based on the name of the variable.

What is the use case for renaming the field ?

**Comment by joprice** [comment]
So far, it's just a curiosity as I'm prototyping but I want to use alternate names for keywords like `class` without having to escape the identifier with backticks, so className instead of ``` ``class`` ```, but I can imagine other cases where the internal and external names differ. Perhaps Emit could be used on the constructor arg? Modifying the example from the docs:

```fsharp
module ParamObject =
      [<AllowNullLiteral>]
      [<Global>]
      type Options [<ParamObject; Emit("$0")>] (
           [<Emit("class")>]
           className: string, ?isCaseSensitive: bool) =
        member val className: string = jsNative with get, set
   ...
```

**Reply by MangelMaxime** [reply]
The main goal for `ParamObject` is to offers a binding to a JavaScript library, for this reason I believe it is best to stay close to the original names used in the library so people don't get too confused and can easily translate the code.
